R/teal_data-extract.R

```diff
@@ -0,0 +1,32 @@
+#'
+#' @section Subsetting:
+#' `x[names]` subsets objects in `teal_data` environment and limit the code to the necessary needed to build limited
+#' objects.
+#'
+#' @param names (`character`) names of objects included in `teal_subset` to subset
+#' @param x (`teal_data`)
+#'
+#' @examples
+#'
+#' # Subsetting
+#' data <- teal_data()
+#' data <- eval_code(data, "a <- 1;b<-2")
+#' data["a"]
+#' data[c("a", "b")]
+#'
+#' join_keys(data) <- join_keys(join_key("a", "b", "x"))
+#' join_keys(data["a"]) # should show empty keys
+#' join_keys(data["b"])
+#' join_keys(data)["a"] # should show empty keys
+#' join_keys(data)["b"]
+#'
+#' @rdname teal_data
+#'
+#' @export
+`[.teal_data` <- function(x, names) {
+  names_no_raw <- names
+  names <- c(names, ".raw_data")
```

Comment:
This is reverse dependency. `teal.data` shouldn't know anything about the `.raw_data`.

Reply:
Alrighty, I will extend `[.qenv` so it allows to pass hidden names
`possible_names <- ls(get_env(x))` -> `possible_names <- ls(get_env(x), all.names = TRUE)`

Will remove `.raw_data` it in `teal.data` and will add `.raw_data` to the names in `teal`

Reply:
- Extended in teal.code https://github.com/insightsengineering/teal.code/pull/216/commits/26efa3fdc6bf42b32c422dba20b3a85d3ce092ab
- Reverted in teal.data https://github.com/insightsengineering/teal.data/pull/346/commits/d629a043d0ab882a388d7375c5f2daa93ca27545
- Updated in teal https://github.com/insightsengineering/teal/pull/1396/commits/e2352b2e025daf89a56533e34f7560586168bd28